torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1844,6 +1848,14 @@ def test_rref_py_pickle_not_supported(self):
             with self.assertRaisesRegex(RuntimeError, "Can not pickle rref in python pickler"):
                 torch.save(local_rref, fname)
 
+    @dist_init
+    def test_remote_throw(self):
+        rref = rpc.remote("worker{}".format((self.rank + 1) % self.world_size),
+                          raise_or_inc,
+                          args=(torch.ones(2),))
+        with self.assertRaisesRegex(Exception, ".*Expected error.*"):
+            rref.to_here()
```

Comment:
Another nice test to have would be to pass this rref to another node and verify calling `rref.to_here()` throws there as well.